godot-cell/tests/mock_blocking.rs

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! A mock implementation of our instance-binding pattern in pure rust for the blocking variant of GdCell.
+//!
+//! Used so we can run miri on this, which we cannot when we are running in itest against Godot.
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::marker::PhantomData;
+use std::sync::{atomic::AtomicUsize, Mutex, OnceLock};
+
+use godot_cell::blocking::{GdCell, InaccessibleGuard};
+
+struct InstanceBinding(*mut ());
+
+unsafe impl Sync for InstanceBinding {}
+unsafe impl Send for InstanceBinding {}
+
+static INSTANCE_BINDINGS: OnceLock<Mutex<HashMap<usize, InstanceBinding>>> = OnceLock::new();
+
+struct InstanceStorage<T> {
+    cell: GdCell<T>,
+}
+
+fn binding() -> &'static Mutex<HashMap<usize, InstanceBinding>> {
+    INSTANCE_BINDINGS.get_or_init(|| Mutex::new(HashMap::new()))
+}
```

Comment:
There is quite a bit of code duplication with symbols from `mock.rs`, do you think you could make those `pub(crate)` and reuse them here?

Also concerns functions and constants like `CALLS` if they are duplicated. Where you need different logic due to blocking, don't try to abstract things though; a bit of duplication is fine in those cases.

Reply:
Everything besides `InstanceBinding` depends in someway on the `GdCell` type inside `InstanceStorage`. I could try to make everything generic over the cell type, but it's getting quite complex. Is it worth the effort?